fix: LEAP-872: LEAP-935: Auto-Annotation fixes

deppp opened the pull request:
# Reject All fix
Fix of small mistake in order to unblock Reject All button (`keys` is a function)

# Auto-Accept Suggestions toggle
It's moved to Bottom Bar, not blocking the view of annotation anymore. Style is uniform with Auto-Annotation toggle. When there are suggestions on a screen this toggle is replaced with Accept All/Reject All buttons, because it looks confusing to have both at the same time.
<img width="322" alt="auto-annotation-suggestions" src="https://github.com/HumanSignal/label-studio/assets/1607227/00e0a8a7-d1bd-4365-a32d-5f3961028a81">

# Loading suggestions...
Spinner is moved from Auto-Accept panel to Bottom Bar. While it's loading the toggle is dimmed for better visibility. But toggle is active because its value will be used only after request is finished, so it can still be useful.
<img width="322" alt="auto-annotation-suggestions-spinner" src="https://github.com/HumanSignal/label-studio/assets/1607227/3e3818c0-5cf4-45bf-8f60-b9110344960f">

Now spinner is displayed for any kind of interactive annotations, not only Image ones:
<img width="322" alt="auto-annotation-spinner" src="https://github.com/HumanSignal/label-studio/assets/1607227/db111b55-ea1e-4823-a525-d4647da9382f">


### PR fulfills these requirements
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

### Describe the reason for change
LEAP-872: Auto-Accept panel is blocking images
LEAP-935: Reject All button not working
Release PR: https://github.com/HumanSignal/label-studio/pull/5715

#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure

Commit:
fixes for LEAP-872, LEAP-935, LEAP-942, including moving the
auto-accept checkbox to the bottom bar, fixing reject all button not
working, and removing the regions use for interactivity

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8625836247

hlomzik committed:
Fix branch name

Gondragos commented on web/libs/editor/src/components/ImageView/SuggestionControls.js:
```diff
@@ -84,14 +84,22 @@ export const SuggestionControls = observer(({ item, useLayer }) => {
         />
         <ControlButton
-          onClick={() => item.annotation.rejectSuggestion(item.id)}
+          onClick={() => {
+            const ann = item.annotation;
+            ann.rejectSuggestion(item.id);
+            ann.deleteAllDynamicregions(true);
+          }}
```
Most probably this is an unintentional change. It changes the current logic instead of fixing LEAP-942. 

When we work with a single suggestion we do not want to affect all other as we may not have a resolution of what we want to do with them on this action. Right now it became duplicate of rejectAllSuggestions which shouldn't be the case.

deppp replied:
It's intentional and follows the following logic:
- dynamic region is always removed after it was sent to the ML Backend
- therefore there could not be the case where you have more than a single dynamic region on the screen unless you use a special tool

Does it match your expectation?

deppp replied:
After conversation with Andrew internally it was indeed identified as not expected and we plan to revert it back.

Gondragos commented on web/libs/editor/src/components/ImageView/SuggestionControls.js:
```diff
@@ -84,14 +84,22 @@ export const SuggestionControls = observer(({ item, useLayer }) => {
           fill="#DD0000"
           iconColor="#fff"
           icon={IconCross}
         />
         <ControlButton
           x={32}
-          onClick={() => item.annotation.acceptSuggestion(item.id)}
+          onClick={() => {
+            const ann = item.annotation;            
+            ann.acceptSuggestion(item.id);
+            ann.deleteAllDynamicregions(true);
+          }}
```
The same as for `rejectSuggestion(item.id)` above

Gondragos commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8692991218)
> Successfully merged:  109 files changed, 2127 insertions(+), 1958 deletions(-)

Gondragos committed:
Merge branch 'develop' into 'fb-ml-release-blocker'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8692991218

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8693015544

hlomzik committed:
Return FF_DEV_1284 back, keeping the fix for Reject All

hlomzik committed:
Tiny fixes (linting mostly + excess styles)

hlomzik committed:
Return back the old behaviour for Accept/Reject

Keep original dynamic regions for 1-by-1 accept and reject.
This is helpful for Segment Anything keypoints case.
https://labelstud.io/blog/get-started-using-segment-anything/#label-with-keypoints

hlomzik committed:
Don't display both auto-accept and accept/reject

They don't have much sense to be displayed simultaneously.
Plus this reduces the space occupied and allows to add more info.

hlomzik committed:
Fix loading state and conditions

- don't display Auto-Accept panel for not supported tags
- display spinner while request is in progress
- also dim the toggle if it's there, but don't disable it

hlomzik committed:
trailing spaces

hlomzik committed:
Little code clean-up

hlomzik committed:
Remove unused component

hlomzik committed:
Actually we need bigger injection - revert

hlomzik reviewed (approved):
kind of cheating, because recent changes are mine, but we have a confidence here and this change is in rc already

hlomzik committed:
Merge branch 'develop' into fb-ml-release-blocker

hlomzik committed:
Rename AutoAcceptToggle from .js to .jsx

to conform recent effort for correct extensions

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8728273698

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8741656982)
> Successfully merged:  24 files changed, 304 insertions(+), 440 deletions(-)

hlomzik committed:
Merge branch 'develop' into 'fb-ml-release-blocker'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8741656982

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8741678799
